### ISSUE TITLE
Inactive optional plugin won't run

### ISSUE THREAD

**Issue by derrickb** [pyblish/pyblish-qml#218]
This could be related to a similar issue I filed on pyblish-lite: https://github.com/pyblish/pyblish-lite/issues/90

I have an optional Validator, the first in the list, that I want to be off by default as it could take a long time to complete for larger scenes.

I have these properties on the plugin:
```
optional = True
active = False
```

If I attempt to run it either by itself or with the other plugins disabled, it doesn't get applied:
![image](https://cloud.githubusercontent.com/assets/1793769/22807414/afd017ce-eedb-11e6-8620-38a1418df05f.png)
![image](https://cloud.githubusercontent.com/assets/1793769/22807418/b3f7e55c-eedb-11e6-8f2f-fc9c6134068d.png)

However if I change "active" to "True", then it does work:
![image](https://cloud.githubusercontent.com/assets/1793769/22807426/be8019f4-eedb-11e6-8ba7-1ddae89c4fdd.png)

**Comment by mottosso** [comment]
Thanks for the detailed report @derrickb. 

This looks to be some problem with the defaulting of the visual toggle in the GUI. `active` is the value the toggle is manipulating, and `optional` is what enables the user to manipulate it. By the looks of it, the toggle isn't getting the initial value of `active` as it is being loaded into the GUI.

Having a look at this as soon as possible.

**Comment by mottosso** [comment]
Which version are you on?

```python
>>> import pyblish_qml
>>> print(pyblish_qml.__version__)
0.7.1
```

**Comment by derrickb** [comment]
0.7.1

**Comment by mottosso** [comment]
This has now been fixed in 0.8.1

For completeness and future self, the issue was that the Pyblish Iterator is using the `active` flag as a means of determining whether or not to process a plug-in, but QML did not update this variable. Since QML runs remotely, it proved tricky to try and modify this member and so instead I implemented a slimmed down version of the iterator specifically for QML.

For now, it's an ok solution, but it does defeat the initial purpose of a common iterator. Namely that if behaviour changes I'll now need to update both base and qml separately.
